### PR TITLE
TDSS Support

### PR DESCRIPTION
**Pull request by lilgreenbird**
this adds TDSS support via a breaking change to encrypt connection property to string type and adding the option "strict"

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/clientcertauth/ClientCertificateAuthenticationTest.java [hunk 1]
```diff
@@ -26,8 +26,7 @@
 
 
 /**
- * Tests client certificate authentication feature
- * The feature is only supported against SQL Server Linux CU2 or higher.
+ * Tests client certificate authentication feature The feature is only supported against SQL Server Linux CU2 or higher.
```
Comment (Jeffery-Wasty) [inline src/test/java/com/microsoft/sqlserver/jdbc/clientcertauth/ClientCertificateAuthenticationTest.java:29]: Missing punctuation. Should be either:

- Tests client certificate authentication feature. The feature is only supported against SQL Server Linux CU2 or higher. OR
- Tests client certificate authentication feature; the feature is only supported against SQL Server Linux CU2 or higher.

Comment (Jeffery-Wasty) [inline src/test/java/com/microsoft/sqlserver/jdbc/clientcertauth/ClientCertificateAuthenticationTest.java:29]: Missing punctuation. Should be either:

- Tests client certificate authentication feature. The feature is only supported against SQL Server Linux CU2 or higher. OR
- Tests client certificate authentication feature; the feature is only supported against SQL Server Linux CU2 or higher.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 2]
```diff
@@ -733,7 +780,8 @@ public String toString() {
             {"server", SQLServerDriverStringProperty.SERVER_NAME.toString()},
             {"domainName", SQLServerDriverStringProperty.DOMAIN.toString()},
             {"port", SQLServerDriverIntProperty.PORT_NUMBER.toString()}};
-    static private final AtomicInteger baseID = new AtomicInteger(0); // Unique id generator for each instance (used for
+    static private final AtomicInteger baseID = new AtomicInteger(0); // Unique id generator for each instance (used
+                                                                      // for
```
Comment (Jeffery-Wasty) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:784]: Combine lines 784 and 785: "// for logging).

Comment (Jeffery-Wasty) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:784]: Combine lines 784 and 785: "// for logging).

**Review by tkyc — APPROVED** [review]

File: src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerDataSource.java [hunk 3]
```diff
@@ -143,18 +152,20 @@
     boolean getTransparentNetworkIPResolution();
 
     /**
-     * Sets a Boolean value that indicates if the trustServerCertificate property is enabled.
+     * Sets a value that indicates if the trustServerCertificate property is enabled.
```
Comment (v-zhangw) [inline src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerDataSource.java:155]: Why remove "boolean"? This method is still setting a boolean value no?

File: src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerDataSource.java [hunk 4]
```diff
@@ -143,18 +152,20 @@
...
      *        true, if the server Secure Sockets Layer (SSL) certificate should be automatically trusted when the
-     *        communication layer is encrypted using SSL. Otherwise, false.
+     *        communication layer is encrypted using SSL. false, if server SLL certificate should not be trusted
+     *        certificate location, if encrypt=strict
      */
     void setTrustServerCertificate(boolean e);
 
     /**
-     * Returns a Boolean value that indicates if the trustServerCertificate property is enabled.
+     * Returns value that indicates if the trustServerCertificate property is enabled.
```
Comment (v-zhangw) [inline src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerDataSource.java:165]: Why remove "boolean"? This method is still setting a boolean value no?

File: src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerDataSource.java (hunk 3 above)
Comment (lilgreenbird) [reply src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerDataSource.java:155]: you are right this WAS going to be changed but we decided against it

**Review by Jeffery-Wasty — APPROVED** [review]

**Review by tkyc — APPROVED** [review]

**Review by v-zhangw — APPROVED** [review]

**Pull request merged**